pfb/writer.py

```diff
@@ -0,0 +1,283 @@
+                                                                "type": {
+                                                                    "type": "map",
+                                                                    "values": "string",
+                                                                },
+                                                            },
+                                                        ],
+                                                    },
+                                                },
+                                            },
+                                        ],
+                                    },
+                                },
+                            },
+                            {
+                                "name": "misc",
+                                "type": {"type": "map", "values": "string"},
+                            },
+                        ],
+                    }
+                ]
+                + encoded_schema,
+            },
+            {
+                "name": "relations",
+                "type": {
+                    "type": "array",
+                    "items": {
+                        "type": "record",
+                        "name": "Relation",
+                        "fields": [
+                            {"name": "dst_id", "type": "string"},
+                            {"name": "dst_name", "type": "string"},
+                        ],
+                    },
+                },
+                "default": [],
+            },
+        ],
+    }
+
+
+class PFBWriter(PFBBase):
+    open_mode = "wb"
+
+    def __init__(self, file_or_path):
+        super(PFBWriter, self).__init__(file_or_path)
+        self._hooks = []
+
+    def copy_schema(self, reader):
+        self.set_schema(deepcopy(reader.schema))
+        self.set_metadata(reader.metadata)
+
+    def write(self, iterable=None, metadata=True):
+        def _iter():
+            if metadata:
+                yield avro_record(None, "Metadata", self._metadata, [])
+            if iterable is not None:
+                for record in iterable:
+                    obj = record["object"]
+                    name = record["name"]
+                    record["object"] = (name, obj)
+                    for hook in self._hooks:
+                        record = hook(record)
+
+                    to_update = {}
+                    for field, value in obj.iteritems():
+                        if value and self.is_base64(name, field):
+                            obj[field] = b64_encode(value)
+                    obj.update(to_update)
+                    yield record
+
+        writer(self._file_obj, make_avro_schema(self.schema), _iter())
+
+    def rename_node(self, name_from, name_to):
+        for node in self.schema:
+            if node["name"] == name_from:
+                node["aliases"] = node.get("aliases", []) + [name_from]
```

Comment:
Should be `node["aliases"] = node.get("aliases", []) + [name_to]`.

Reply:
I saw your fix in https://github.com/uc-cdis/pypfb/commit/7be5abeff728cc9275a7c59f9d6b65924a29fcf1#diff-447ae0df07b4135197472a08ac2dcc3cR155, but why is that, please?

Reply:
Cuz the function is renaming the node. The original name is `name_from`, when the node name is renamed to `name_to` the aliases should be updated accordingly